File: restaurant/strategy/payment_strategy.py
from abc import ABC, abstractmethod

class PaymentStrategy(ABC):
    """Abstract base class cho chiến lược thanh toán"""
    
    @abstractmethod
    def pay(self, amount: float) -> bool:
        """Thực hiện thanh toán"""
        pass
    
    @abstractmethod
    def refund(self, amount: float) -> bool:
        """Hoàn tiền"""
        pass
    
    @abstractmethod
    def get_payment_method_name(self) -> str:
        """Lấy tên phương thức thanh toán"""
        pass

File: backend/src/main/java/com/restaurant/pattern/strategy/PaymentStrategy.java
package com.restaurant.pattern.strategy;

/**
 * Interface cho Strategy Pattern - đại diện cho các phương thức thanh toán
 */
public interface PaymentStrategy {
    /**
     * Thực hiện thanh toán
     *
     * @param amount Số tiền cần thanh toán
     * @return True nếu thanh toán thành công, False nếu thất bại
     */
    boolean pay(double amount);

    /**
     * Lấy tên phương thức thanh toán
     */
    String getPaymentMethod();
}
